CRM/Contact/Form/Task/PDFLetterCommon.php

```diff
@@ -442,34 +497,62 @@ public static function createActivities($form, $html_message, $contactIds) {
     if (!empty($form->_activityId)) {
       $activityParams += array('id' => $form->_activityId);
     }
-    if ($form->_cid) {
-      $activity = CRM_Activity_BAO_Activity::create($activityParams);
-      if (!empty($form->_caseId)) {
-        $caseActivityParams = array('activity_id' => $activity->id, 'case_id' => $form->_caseId);
-        CRM_Case_BAO_Case::processCaseActivity($caseActivityParams);
-      }
+
+    // This seems silly, but the old behavior was to first check `_cid`
+    // and then use the provided `$contactIds`. Probably not even necessary,
+    // but difficult to audit.
+    $contactIds = $form->_cid ? array($form->_cid) : $contactIds;
+
+    $activityContacts = CRM_Core_OptionGroup::values('activity_contacts', FALSE, FALSE, FALSE, NULL, 'name');
```

Comment:
Ironically this is one of the functions that I think should be deprecated - in this case by calling the api rather than wrangling out the variables first